autoload/db/adapter/bigquery.vim

```diff
@@ -0,0 +1,81 @@
+if !executable('bq')
+  echom 'bq command is not available.'
+  let g:autoloaded_db_bigquery = 0
+endif
+
+if exists('g:autoloaded_db_bigquery')
+  finish
+endif
+
+let g:autoloaded_db_bigquery = 1
+
+function! s:command_for_url(params, action) abort
+  let cmd = ['bq']
+  let subcmd = [a:action]
+
+  for [k, v] in items(a:params)
+    let op = ' --'.k.'='.v
+
+    if len(matchstr(k, 'legacy'))
+        " FIXME: parse subcmd options
+        call add(subcmd, op)
+    else
+        " parse global options
+        call add(cmd, op)
+    endif
+
+  endfor
+
+  return join(cmd + subcmd)
+
+endfunction
+
+function! s:params(url) abort
+  let parsed_params = db#url#parse(a:url)
+  let conn_params = parsed_params.params
+  return conn_params
+endfunction
+
+function! db#adapter#bigquery#interactive(url, action) abort
+  echom "Starting bigquery"
+  return split(s:command_for_url(s:params(a:url), a:action))
+endfunction
+
+function! db#adapter#bigquery#input(url, in) abort
+
+  " QST: what is the purpose of input?
+  let qry = join(readfile(a:in, 2))
+  if len(qry)
+    " QST: why I have to add this condition?
+    return db#adapter#bigquery#interactive(a:url, "query") + [qry]
+  endif
+
+  return []
+
+endfunction
+
+function! db#adapter#bigquery#complete_opaque(url) abort
+  " QST: don't know why this is required
```

Comment:
"Opaque" describes URLs like `mailto:user@example.com`, which do not adhere to the standard `schema://` pattern. Most Dadbod adapters accept both `adapter:dbname` and `adapter:///dbname`, which is why you'll commonly find one method delegating to the other.

Reply:
I see, then I can drop it, can't I?